org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend

```diff
@@ -456,12 +460,28 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 				
 				@�Inject�
 				protected void init(�IGrammarAccess� access) {
+					�IF partitions.size > 1�
+						�FOR partition : partitions.indexed�
+							init�partition.key�(access);
+						�ENDFOR�
+					�ELSE�
+						grammarAccess = (�grammar.grammarAccess�) access;
+						�FOR group : allAmbiguousTransitionsBySyntax�
+							match_�group.identifier� = �group.elementAlias.elementAliasToConstructor�;
+						�ENDFOR�
+					�ENDIF�
+				}
+			  
+			  �FOR partition : partitions.indexed�
```

Comment:
init0.... should be generated only if partions size > 1

Reply:
I added the if statement